File: day1/puzzle2.ts
import { readInput } from '../utils';

const input = readInput('./input').map((i) => Number(i));

console.log(
  input
    .map((x, i) => (i < input.length - 2 ? Number(x < input[i + 3]) : 0))
    .reduce((x, y) => x + y)
);

File: 2022/day3/puzzle2.rb
def get_badge (group)
  group.map(&:chars).reduce(&:intersection).first
end

def get_priority (char)
  if char in 'a'..'z'
    char.ord - 'a'.ord + 1
  else
    char.ord - 'A'.ord + 27
  end
end


def solve(input)
  bags = input.split("\n")

  groups = bags.each_slice(3)

  groups.map(&method(:get_badge)).map(&method(:get_priority)).sum
end

puts solve(File.read('./input'))
